cashu/wallet/api/router.py

```diff
@@ -127,6 +128,46 @@ async def invoice(
     return
 
 
+@router.post(
+    "/swap",
+    name="Multi-mint swaps",
+    summary="Swap funds between mints.",
+    response_model=SwapResponse,
+)
+async def swap(
+    amount: int = Query(default=..., description="Amount to swap between mints"),
+    outgoing_mint: str = Query(default=..., description="URL of outgoing mint"),
+    incoming_mint: str = Query(default=..., description="URL of incoming mint"),
+):
+    if not settings.lightning:
+        raise Exception("this mint does not support lightning.")
+    if outgoing_mint in incoming_mint:
+        raise Exception("mints for swap have to be different")
+    # request invoice from incoming mint
+    wallet = create_wallet(url=incoming_mint)
+    invoice = await wallet.request_mint(amount)
+    # pay invoice from outgoing mint
+    wallet = await load_mint(wallet, outgoing_mint)
```

Comment:
bit of a nitpick, but `incoming_wallet` and `outgoing_wallet` would be a lot clearer to follow than shadowing `wallet` for both, as two calls to `create_wallet` rather than using `load_mint`

```suggestion
    # request invoice from incoming mint
    incoming_wallet = create_wallet(url=incoming_mint)
    outgoing_wallet = create_wallet(url=outgoing_mint)
    invoice = await incoming_wallet.request_mint(amount)
    # pay invoice from outgoing mint
    await outgoing_wallet.load_proofs()
    total_amount, fee_reserve_sat = await outgoing_wallet.get_pay_amount_with_fees(invoice.pr)
```